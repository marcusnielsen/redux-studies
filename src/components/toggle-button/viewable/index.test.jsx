import React from 'react'
import {createStore} from 'redux'
import {mount} from 'enzyme'
import ToggleButton from '../../toggle-button'
import Viewable from '../viewable'

test('viewable', function onTest () {
  const initialState = {
    active: false
  }

  const provider = {
    dispatch: null
  }

  const model = ToggleButton({
    entity: 'a',
    id: 'b',
    field: 'c',
    initialState,
    provider
  })

  const viewable = Viewable({model})
  expect(Object.keys(viewable)).toMatchSnapshot()
})


test('viewable.view update interaction', function onTest () {
  let dispatchProxy = null

  function dispatch (data) {
    dispatchProxy(data)
  }

  const initialState = {
    active: false
  }

  const provider = {
    dispatch
  }

  const model = ToggleButton({
    entity: 'a',
    id: 'b',
    field: 'c',
    initialState,
    provider
  })

  const store = createStore(model.rootReducer)
  dispatchProxy = store.dispatch

  const viewable = Viewable({model})
  const ReactView = viewable.view
  const wrapper = mount(<ReactView active={false} />)
  // @TODO: continue here.
  expect(wrapper.simulate('click').html()).toMatchSnapshot()
})
