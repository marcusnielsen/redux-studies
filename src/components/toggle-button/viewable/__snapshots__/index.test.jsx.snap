exports[`test viewable 1`] = `
Array [
  "actionTypes",
  "actions",
  "rootReducer",
  "entity",
  "id",
  "field",
  "initialState",
  "propTypes",
  "view",
]
`;

exports[`test viewable.view update interaction 1`] = `"<div>false</div>"`;
